test/Microsoft.AspNet.Hosting.Tests/WebHostBuilderTests.cs

```diff
@@ -137,22 +137,27 @@ public void UseEnvironmentIsNotOverriden()
             var config = builder.Build();
 
             var expected = "MY_TEST_ENVIRONMENT";
-            var webHost = new WebHostBuilder(config, captureStartupErrors: true).UseEnvironment(expected).Build();
+            var application = new WebApplicationBuilder()
+                .UseConfiguration(config)
+                .UseEnvironment(expected)
+                .UseServer(new TestServer())
+                .Build();
 
-            Assert.Equal(expected, webHost.ApplicationServices.GetService<IHostingEnvironment>().EnvironmentName);
+            Assert.Equal(expected, application.Services.GetService<IHostingEnvironment>().EnvironmentName);
         }
 
-        private WebHostBuilder CreateWebHostBuilder()
+        private WebApplicationBuilder CreateWebHostBuilder()
         {
             var vals = new Dictionary<string, string>
             {
                 { "server", "Microsoft.AspNet.Hosting.Tests" },
                 { "DetailedErrors", "true" },
+                { "captureStartupErrors", "true" }
             };
             var builder = new ConfigurationBuilder()
                 .AddInMemoryCollection(vals);
             var config = builder.Build();
-            return new WebHostBuilder(config, captureStartupErrors: true);
+            return new WebApplicationBuilder().UseConfiguration(config).UseServer(new TestServer());
```

Comment:
This is very strange, there is no ServerFactory in this assembly, and most of the tests override it manually anyways by setting the TestServer explicitly.